src/ros/tf/republisher.cpp

```diff
@@ -137,11 +137,12 @@ void TFRepublisher::create_cursor(double start_time)
 	if(cursor_!=NULL) {
 		mongoc_cursor_destroy(cursor_);
 	}
-	if(collection_) {
-		delete collection_;
-	}
-	collection_ = MongoInterface::get_collection(
-		db_name_.c_str(),db_collection_.c_str());
+    if(collection_) {
+        collection_.reset();
```

Comment:
reset is not needed, just remove it everywhere. smart pointer deletes itself, no reset needed.